bigframes/core/blocks.py

```diff
@@ -1853,6 +1855,176 @@ def transpose(
             .with_transpose_cache(self)
         )
 
+    def _generate_sequence(
+        self,
+        start,
+        stop,
+        step: int = 1,
+    ):
+        range_expr = self.expr.from_range(
+            start,
+            stop,
+            step,
+        )
+
+        return Block(
+            range_expr,
+            column_labels=["min"],
+            index_columns=[],
+        )
+
+    def _generate_resample_label(
+        self,
+        rule: str,
+        closed: Optional[Literal["right", "left"]] = None,
+        label: Optional[Literal["right", "left"]] = None,
+        on: Optional[Label] = None,
+        level: typing.Union[LevelType, typing.Sequence[LevelType]] = None,
+        origin: Union[
+            Union[pd.Timestamp, datetime.datetime, numpy.datetime64, int, float, str],
+            Literal["epoch", "start", "start_day", "end", "end_day"],
+        ] = "start_day",
+    ) -> Block:
+        # Validate and resolve the index or column to use for grouping
+        if on is None:
+            if len(self.index_columns) == 0:
+                raise TypeError("Index type not valid. Expected Datetime Type.")
+            if len(self.index_columns) > 1 and (level is None):
+                raise ValueError(
+                    "Multiple indices are not supported for this operation"
+                    " when 'level' is not set."
+                )
+            level = level or 0
+            col_id = self.index.resolve_level(level)[0]
+            # Reset index to make the resampling level a column, then drop all other index columns.
+            # This simplifies processing by focusing solely on the column required for resampling.
+            block = self.reset_index(drop=False)
```

Comment:
I think this might require a total ordering in BigFrames to work. Could you run some tests with unordered mode + a timestamp index that has non-unique values?

Reply:
Added a test for unordered mode and with duplicate values, seems runs fine. I don't think the order matters here, in the end we are not keeping this order.